test/sql/alter/add_pk/test_add_pk_rollback.test

```diff
@@ -25,7 +25,7 @@ INSERT INTO test VALUES (2, 1);
 statement error
```

Comment:
Here is a list of issues that might be related to this PR, could you perhaps look and see if these now work/if not why not - and add the tests?

https://github.com/duckdb/duckdb/issues/11288
https://github.com/duckdb/duckdb/issues/11915
https://github.com/duckdb/duckdb/issues/8764
https://github.com/duckdb/duckdb/issues/7206
https://github.com/duckdb/duckdb/issues/7182
https://github.com/duckdb/duckdb/issues/6738
https://github.com/duckdb/duckdb/issues/6500
https://github.com/duckdb/duckdb/issues/5807
https://github.com/duckdb/duckdb/issues/1631
https://github.com/duckdb/duckdb/issues/3249
https://github.com/duckdb/duckdb/issues/3265
https://github.com/duckdb/duckdb/issues/4023
https://github.com/duckdb/duckdb/issues/4214
https://github.com/duckdb/duckdb/issues/4886

Reply:
- [x] #1288
- [x] #8764
- [x] #7182
- [x] #6500
- [x] #5807
- [x] #5771 
- [x] #4886
- [x] #4214 
- [x] #1631

I have not included the following specifically, as they are already covered by other included issues.
- #4023, #3265, #3249, #3249, #6738, #7206, #11915

I'll also have another pass over issues that we've potentially missed yesterday.

Reply:
More resolved issues:
- #11102
- #12684
- #8147
- #6787
- #4807